main/src/main/scala/sbt/Defaults.scala

```diff
@@ -264,15 +264,6 @@ object Defaults extends BuildCommon {
           Clean.deleteContents(tempDirectory, _ => false)
       },
       useSuperShell :== sbt.internal.TaskProgress.isEnabled,
-      progressReports := { (s: State) =>
```

Comment:
I consider progressReports to be somewhat stable actually.
super shell is one implementation on it.

Reply:
One thing I might change about the `progressReports` is maybe dropping `s: State` and just making it a Seq of TaskProgress.

Reply:
Aah, ok. That makes sense. I understand what the key is better now than when I first wrote the commit. It was in fact the State parameter that was especially throwing me off.